meta-resin-common/recipes-support/resinhup/resinhup/run-resinhup.sh

```diff
@@ -309,6 +313,127 @@ function setCurrentVersion() {
 	fi
 }
 
+# When updating, check if the target resinOS version has newer default supervisor
+# than what is running on the device at the moment. If there is, fill out the
+# UPDATER_SUPERVISOR_TAG parameter, plus some some helper parameters so not to
+# redo work in the next step (pulling the supervisor)
+function getSupervisorVersionFromRelease() {
+    if [ "$STAGING" == "yes" ]; then
+        DEFAULT_SUPERVISOR_VERSION_URL_BASE="https://s3.amazonaws.com/resin-staging-img/"
+    else
+        DEFAULT_SUPERVISOR_VERSION_URL_BASE="https://s3.amazonaws.com/resin-production-img-cloudformation/"
+    fi
+    DEFAULT_SUPERVISOR_VERSION_URL="${DEFAULT_SUPERVISOR_VERSION_URL_BASE}images/${SLUG}/${HOSTOS_VERSION}/VERSION"
+
+    # Get supervisor version for target resinOS release, it is in format of "a.b.c-shortsha", e.g. "4.1.2-f566dc4dd241",
+    # and tag new version for the device if it's newer than the current version, from the API
+    DEFAULT_SUPERVISOR_VERSION=$(curl -s "$DEFAULT_SUPERVISOR_VERSION_URL" | sed 's/-[[:xdigit:]]*//')
+    if [ -z "$DEFAULT_SUPERVISOR_VERSION" ] || [ -z "${DEFAULT_SUPERVISOR_VERSION##*xml*}" ]; then
+        log ERROR "Could not get the default supervisor version for this resinOS release, skipping update."
```

Comment:
The log level doesn't match the message. I would weak the message to say that it actually stops performing the update if we could not get the default supervisor version.